### PR TITLE
Switch postgres & set deployment mechanism

### PR DESCRIPTION
**Pull request by rafaparadela**
This PR: 

- Fixes: #149 Switch to the Postgres database for persistence
- Fixes: #151 Prepare deployment environment for version 2

@dialelo @juanpedromoreno @raulraja Could you review please?

_Notes:_
- Command to deploy: 
```bash
sbt clean stage deployHeroku
```

- Command to monitor logs: 
```bash
heroku logs --app scala-exercises --tail
```

- Prod. URL: http://scala-exercises.herokuapp.com/

### CODE REVIEW

**Comment by raulraja** [comment]
LGTM! Let's get this in so we can setup PR review apps soon! Great stuff, thanks for doing this so quickly!

**Comment by rafaparadela** [comment]
I've been trying to deploy it with a simple `git push heroku master` but it doesn't detect the `Procfile`. It seems like we have to define it in the `server` module level.

File: site/server/conf/evolutions/default/1.sql [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-    id bigint primary key auto_increment,
-    login varchar(255) unique not null,
-    name varchar(255) not null,
-    githubId varchar(255) not null,
-    pictureUrl varchar(255) not null,
-    githubUrl varchar(255) not null,
-    email varchar(255) not null
+CREATE TABLE users (
+    id bigserial primary key,
+    login text UNIQUE NOT NULL,
```
Comment (dialelo) [inline site/server/conf/evolutions/default/1.sql]: Doesn't Doobie complain about materializing `text` fields into a Scala `String`?

**Comment by dialelo** [comment]
Other than the question about converting `text` fields to strings LGTM, great work!

**Comment by dialelo** [comment]
From a conversation with @juanpedromoreno: We can still use the name `User` for the user table although the name has to be quoted in the queries, like the following: `SELECT * FROM "User"`

File: site/server/conf/evolutions/default/1.sql (hunk 1 above)
Comment (rafaparadela) [inline site/server/conf/evolutions/default/1.sql]: it doesn't. It only complains when weren't defined as 'NOT NULL'. Tests are passing successfully.

**Pull request merged**